### PR TITLE
Removing comments for python2 in code printers and codegen

### LINKED ISSUE #17082: Deprecate python 2 support in code printers and and python codegen
See #16901
Remove `standard='python2'` for `render_as_module` and `pycode` and `PythonCodePrinter`

### PR DESCRIPTION
**Pull request by Smit-create**
Removing python2 in code printers and codegen

#### References to other Issues or PRs

Fixes #17082

#### Backwards compatibility breaks and deprecations
`Python2` support has removed from `PythonCodePrinter`which provided options to switch between `Python2` and `Python3`. After deprecation, it supports for `Python3` only. 

#### Release Notes

* printing
    * Removing python2 support from PythonCodePrinter

### CODE REVIEW

**Comment by Smit-create** [comment]
@sylee957  Please review this.

**Comment by czgdp1807** [comment]
IMO, this PR doesn't need a release notes entry as python 2 support has already been notified in some earlier PRs.

**Comment by moorepants** [comment]
I'm not sure that removing the ability to generate Python 2 code should be part of SymPy dropping support wrt to depending on Python 2. These are different things. For example, we support generating old and new versions of C.

If people agree this should be removed, then these changes will need to follow the deprecation policy. You are removing keyword arguments that change the public API.

**Comment by Smit-create** [comment]
@moorepants  I edited the OP.  Can you please suggest what further changes are required? Or removing `python2` support is not good at present ?

**Comment by sylee957** [comment]
It was my PR in #16901 I had added `'python2'` standard.
I have just used the elaborated word `'standard'` instead of adding it as an arbitrary feature, but https://github.com/sympy/sympy/pull/16901#issuecomment-503301689 and https://github.com/sympy/sympy/pull/16901#issuecomment-507028121.

But, I think that it may not be a good feature to add because it complicates things like `_print_Pow`. And one other argument is that we may not be able to test it in the CI.
I would raise a deprecation warning as it had passed the previous release.

However, if someone else comments that a feature for python 2 code printing is need for a special purpose, I would change my stance to maintain it.

File: sympy/codegen/pyutils.py [hunk 1]
```diff
@@ -3,18 +3,17 @@
...
     """Renders python code as a module (with the required imports)
 
     Parameters
     ==========
 
-    standard
-        See the parameter ``standard`` in
-        :meth:`sympy.printing.pycode.pycode`
+    content: python code
+        The python code which is to be rendered as module
```
Comment (oscarbenjamin) [inline sympy/codegen/pyutils.py]: I think it should be `content: string` (assuming it is a string...)

**Comment by asmeurer** [comment]
Adding a flag only to immediately deprecate it seems odd. I personally am not opposed to removing it outright, but I'm also not opposed to keeping it around. It doesn't seem like it results in too much extra code.

**Comment by asmeurer** [comment]
Oh I see, `standard='python2'` was already there. In that case we definitely could deprecate it if we want to remove it. But I personally think we should just keep it.

**Comment by oscarbenjamin** [comment]
> Oh I see, `standard='python2'` was already there. In that case we definitely could deprecate it if we want to remove it. But I personally think we should just keep it.

I think what @sylee957 is saying is that it was added as a fudge to pass the py2 tests while improving the output on py3. It is not really a complete feature though. Since it will probably never be completed it is probably not worth having as it is likely that the rest of the code will become incompatible with Python 2 over time anyway now that py2 compatibility isn't tested.

This feature has only been in one release (1.5) and had a note in the docstring saying "this parameter may be removed in the future". I doubt that anyone anywhere is using it.

**Comment by asmeurer** [comment]
I don't feel strongly about it either way, whether it should be removed or kept.

**Comment by oscarbenjamin** [comment]
Reading the comments above I would summarise the situation from reviewers like this:

1. @asmeurer feels that we shouldn't remove this just for the sake of it but also doesn't feel strongly about it either way.
2. @sylee957 would like to remove this but suggests that deprecation is necessary. That means we need to make the code give a warning and then it can be deleted later.
3. I would vote to merge the pull request and delete this right now. I very much doubt that anyone is using this (or will be by the time of the next release). I expect that leaving this feature there for a few releases will create more confusion than simply removing it now.

**Comment by czgdp1807** [comment]
IMO, we should remove without deprecation as it was made clear that `SymPy 1.6` will not support `Python 2`. It looks like ready to merge.

**Comment by moorepants** [comment]
> SymPy 1.6 will not support Python 2

Just to be clear, this is not a Python 2 support issue. We print code for multiple languages and different language versions. You can still print Python 2 code if SymPy can't run in Python 2.

**Comment by czgdp1807** [comment]
Thanks @moorepants for pointing out. So may be this PR can be closed. Attaching a `Could Close` label to it.

**Comment by sylee957** [comment]
Maybe the XXX comments I have added before can be removed instead, if it is going in this direction.
I think that there are places where python2 syntax is continued forever, as I see pypy.

**Comment by Smit-create** [comment]
Okay!, I will remove those comments.

**Comment by Smit-create** [comment]
Following the above discussion, I just removed the comments specifying the removal of python 2 support.

**Pull request merged**